compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -3102,51 +3380,81 @@ abstract class GenJSCode extends plugins.PluginComponent
 
       val boxedResult = sym.name match {
         case JSUnaryOpMethodName(code) if argc == 0 =>
+          requireNotSuper()
           js.JSUnaryOp(code, receiver)
 
         case JSBinaryOpMethodName(code) if argc == 1 =>
+          requireNotSuper()
           js.JSBinaryOp(code, receiver, args.head)
 
-        case nme.apply if receiver0.tpe.typeSymbol.isSubClass(JSThisFunctionClass) =>
+        case nme.apply if sym.owner.isSubClass(JSThisFunctionClass) =>
+          requireNotSuper()
           js.JSBracketMethodApply(receiver, js.StringLiteral("call"), args)
 
         case nme.apply if !hasExplicitJSEncoding =>
+          requireNotSuper()
           js.JSFunctionApply(receiver, args)
 
         case _ =>
-          def jsFunName = jsNameOf(sym)
+          def jsFunName = js.StringLiteral(jsNameOf(sym))
+
+          def genSelectGet(propName: js.Tree): js.Tree = {
+            superIn.fold[js.Tree] {
+              js.JSBracketSelect(receiver, propName)
+            } { superInSym =>
+              js.JSBracketSuperSelect(
+                  jstpe.ClassType(encodeClassFullName(superInSym)),
+                  receiver, propName)
+            }
+          }
+
+          def genSelectSet(propName: js.Tree, value: js.Tree): js.Tree = {
+            val lhs = superIn.fold[js.Tree] {
+              js.JSBracketSelect(receiver, propName)
+            } { superInSym =>
+              js.JSBracketSuperSelect(
+                  jstpe.ClassType(encodeClassFullName(superInSym)),
+                  receiver, propName)
+            }
```

Comment:
Ah ah, true, they *do* the same thing. Although it's semantically completely unrelated, which I guess is why I did not "see" the duplication. I'll add a third `def` to delegate to in `genSelect{Get,Set}`.